examples/step-32/step-32.cc

```diff
@@ -2716,26 +2716,31 @@ namespace Step32
     using CellFilter =
       FilteredIterator<typename DoFHandler<2>::active_cell_iterator>;
 
-    WorkStream::run(
-      CellFilter(IteratorFilters::LocallyOwnedCell(),
-                 temperature_dof_handler.begin_active()),
-      CellFilter(IteratorFilters::LocallyOwnedCell(),
-                 temperature_dof_handler.end()),
-      std::bind(&BoussinesqFlowProblem<dim>::local_assemble_temperature_rhs,
-                this,
-                global_T_range,
-                maximal_velocity,
-                global_entropy_variation,
-                std::placeholders::_1,
-                std::placeholders::_2,
-                std::placeholders::_3),
-      std::bind(
-        &BoussinesqFlowProblem<dim>::copy_local_to_global_temperature_rhs,
-        this,
-        std::placeholders::_1),
-      Assembly::Scratch::TemperatureRHS<dim>(
-        temperature_fe, stokes_fe, mapping, quadrature_formula),
-      Assembly::CopyData::TemperatureRHS<dim>(temperature_fe));
+    auto worker =
+      [=](const typename DoFHandler<dim>::active_cell_iterator &cell,
+          Assembly::Scratch::TemperatureRHS<dim> &              scratch,
+          Assembly::CopyData::TemperatureRHS<dim> &             data) {
+        this->local_assemble_temperature_rhs(global_T_range,
+                                             maximal_velocity,
+                                             global_entropy_variation,
+                                             cell,
+                                             scratch,
+                                             data);
+      };
+
+    auto copier = [=](const Assembly::CopyData::TemperatureRHS<dim> &data) {
```

Comment:
You really just copy `this` here, and you've explicitly listed it in other places as well, so let's go with that.
```suggestion
    auto copier = [this](const Assembly::CopyData::TemperatureRHS<dim> &data) {
```